apps/files/js/fileactions.js

```diff
@@ -679,12 +679,19 @@
 				permissions: OC.PERMISSION_READ,
 				icon: '',
 				actionHandler: function (filename, context) {
-					var dir = context.$file.attr('data-path') || context.fileList.getCurrentDirectory();
+					let dir, id
+					if (context.$file) {
+						dir = context.$file.attr('data-path')
+						id = context.$file.attr('data-id')
+					} else{
+						dir = context.fileList.getCurrentDirectory()
+						id = context.objectId
```

Comment:
I'd prefer to use fileId instead of objectId
```suggestion
						id = context.fileId
```